### PR TITLE
Add support for checking AMD HW configuration MSR

### PR DESCRIPTION
**Pull request by superm1**
The MSR 0xc0010015 has two bits that indicate whether the SMM is locked down.

These are checked by chipsec.

Add the same check to fwupd.

Link: https://github.com/chipsec/chipsec/pull/2343

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: docs/hsi-tests.d/org.fwupd.hsi.Amd.SmmLocked.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "id": "org.fwupd.hsi.Amd.SmmLocked",
+  "name": "SMM locked down",
+  "description": [
+    "Disable SMM_BASE from being saved to and restored from the SMM save state area.",
+    "code in the ASeg and TSeg range and the SMM registers are Read-only and SMI interrupts are not intercepted in SVM."
```
Comment (hughsie) [inline docs/hsi-tests.d/org.fwupd.hsi.Amd.SmmLocked.json:6]: Upper case C?

File: docs/hsi-tests.d/org.fwupd.hsi.Amd.SmmLocked.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  ],
+  "failure-impact": [
+    "An attacker with an existing unrelated vulnerability can the SMM save state area to exfiltrate data."
+  ],
+  "failure-results": {
+    "not-supported": "SMM is not locked down"
+  },
+  "success-results": {
+    "supported": "SMM is locked down"
```
Comment (hughsie) [inline docs/hsi-tests.d/org.fwupd.hsi.Amd.SmmLocked.json:15]: I think these are supposed to be enums, e.g. `locked'

File: src/fu-security-attr-common.c [hunk 3]
```diff
@@ -404,6 +408,10 @@ fu_security_attr_get_title(FwupdSecurityAttr *attr)
 		/* TRANSLATORS: Title: Whether BIOS Firmware updates is enabled */
 		return _("BIOS Firmware Updates");
 	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_AMD_SMM_LOCKED) == 0) {
+		/* TRANSLATORS: Title: Whether firmware is locked down */
+		return g_strdup(_("System management mode"));
```
Comment (hughsie) [inline src/fu-security-attr-common.c:413]: for _title, each Word Is Title Cased pls

File: path [hunk 4]
```diff
@@ -0,0 +1,39 @@
+--- docs/hsi-tests.d/org.fwupd.hsi.Amd.SmmLocked.json
```
Comment (qhill) [inline path:1]: I'm not sure you meant to commit this file.

Comment (superm1) [reply path:1]: Whoops thanks!

**Review by hughsie — APPROVED** [review]

**Pull request merged**